File: base/src/main/java/com/example/base/network/base/BaseViewModel.java
package com.example.base.network.base;

import android.app.Application;
import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.MutableLiveData;

/**
 * create by libo
 * create on 2020/8/4
 * description viewModel基类
 */
public class BaseViewModel extends AndroidViewModel {
    /** 请求成功通知 */
    protected MutableLiveData<Boolean> loadSuccess = new MutableLiveData<>();
    /** 请求失败通知 */
    protected MutableLiveData<Boolean> loadFail = new MutableLiveData<>();

    public BaseViewModel(@NonNull Application application) {
        super(application);
    }

    public MutableLiveData<Boolean> getLoadSuccess() {
        return loadSuccess;
    }

    public MutableLiveData<Boolean> getLoadFail() {
        return loadFail;
    }
}

File: library-base/src/main/java/com/example/base/network/base/viewmodel/BaseViewModel.kt
package com.example.base.network.base.viewmodel

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import com.example.base.network.base.livedata.BaseLiveData

/**
 * create by apple
 * create on 2020/8/9
 * description 项目所有viewModel基类
 */
open class BaseViewModel(application: Application) : AndroidViewModel(application) {
    var baseLiveData = BaseLiveData()


}